File: ide-features-trainer/src/training/learn/LessonListenerAdapter.java
package training.learn;

import training.learn.exceptons.BadModuleException;
import training.learn.exceptons.BadLessonException;
import training.learn.exceptons.LessonIsOpenedException;

import java.awt.*;
import java.io.IOException;
import java.util.concurrent.ExecutionException;

/**
 * Created by karashevich on 27/02/15.
 */
public class LessonListenerAdapter implements LessonListener {

    public void lessonStarted(Lesson lesson){
    }
    @Override

    public void lessonPassed(Lesson lesson){

    }

    @Override
    public void lessonClosed(Lesson lesson){

    }

    @Override
    public void lessonNext(Lesson lesson) throws BadLessonException, ExecutionException, IOException, FontFormatException, InterruptedException, BadModuleException, LessonIsOpenedException {
    }

}

File: ide-features-trainer/src/training/learn/lesson/LessonListenerAdapter.kt
package training.learn.lesson

import training.learn.exceptons.BadLessonException
import training.learn.exceptons.BadModuleException
import training.learn.exceptons.LessonIsOpenedException
import training.learn.interfaces.Lesson
import java.awt.FontFormatException
import java.io.IOException
import java.util.concurrent.ExecutionException

open class LessonListenerAdapter : LessonListener {

  override fun lessonStarted(lesson: Lesson) { }
  override fun lessonPassed(lesson: Lesson) { }
  override fun lessonClosed(lesson: Lesson) { }

  @Throws(BadLessonException::class, ExecutionException::class, IOException::class, FontFormatException::class, InterruptedException::class,
          BadModuleException::class, LessonIsOpenedException::class)
  override fun lessonNext(lesson: Lesson) { }

}